Fixes 228 and 229

Anders-Hopkins opened the pull request:
Added gage stats side bar with station type and agencies as parameters. A table is created based off of station type parameter

Anders-Hopkins committed:
GS Sidebar

Added drop down menus to the sidebar for the gagestats page.

Anders-Hopkins committed:
Region Displays

Added a region identifier in the mainpage which changes according the the gagestats sidebar

Anders-Hopkins committed:
Begin Table

Started to build gage table

Anders-Hopkins committed:
Merge branch 'dev' into AH-228

Anders-Hopkins committed:
Added gage sidebar & table

Gagestats sidebar is created and table is displayed by selecting Station Type parameter. Agency parameter does not filter table (yet)

Anders-Hopkins committed:
Merge branch 'dev' into AH-228

Anders-Hopkins committed:
Cleaned up code

Removed gagestats region option

Anders-Hopkins committed:
Merge branch 'dev' into AH-228

Anders-Hopkins committed:
Merged with Dev

Dev branch merged into AH-228

kjacobsen16 commented on src/assets/config.json:
```diff
@@ -4,13 +4,15 @@
     "agenciesURL": "Agencies",
     "citationURL": "Citations",
     "errorsURL": "Errors",
+    "gageURL": "https://test.streamstats.usgs.gov/gagestatsservices/",
```
Since the gageStatsBaseURL was added earlier, you can remove this line. You can do a full Find and Replace in VSCode.

kjacobsen16 commented on src/app/app.component.html:
```diff
@@ -56,13 +56,15 @@
 </nav>
-
+<!-- <wim-sidebar *ngIf="router.url.indexOf('/settings') === -1 && router.url.indexOf('/profile') === -1 && router.url.indexOf('/gagestats') === -1"></wim-sidebar> -->
+<gs-sidebar *ngIf="router.url.indexOf('/gagestats') > -1"></gs-sidebar>
```
To keep consistent with style changes, we should maybe move this to after line 63. That'll help with styling.

kjacobsen16 commented on src/app/gagestats/gagestats.component.ts:
```diff
@@ -8,18 +16,82 @@ import { GagestatsService } from './gagestats.service';
+           });
+      }
+    });
+    */
```
Is this necessary to keep in? Maybe comment why you're keeping it if so?

kjacobsen16 commented on src/app/gs-sidebar/gs-sidebar.component.ts:
```diff
@@ -0,0 +1,89 @@
+              }
+          } else { this.selectedStationType = []; }
+      }*/
+    });
```
Same as above, is this going to be used in the future?

kjacobsen16 reviewed (changes requested):
Left some comments. I mostly just have some cleanup suggestions before merging it in: (1) Mitch pushed the styling changes, so now the components are using scss instead of css, and you can probably remove some of this styling to stay consistent with what Mitch added (he can probably help you with that, he reformatted a lot of it), (2) for the sidebar dropdowns, I recommend using the `ss-multiselect-dropdown` used in the main sidebar component, then you can have the "Check all" and "Uncheck all" options as well, (3) also keep an eye out for cleanup, or make notes of why you're commenting out sections (rather than deleting it) - easiest way is to look through your [changes in the pull request](https://github.com/USGS-WiM/NSS/pull/247/files)

Other than that, it's a great start :)

Anders-Hopkins committed:
Close #228 and #229

Made suggested changes, added .scss files, cleaned up code

Anders-Hopkins committed:
Merge branch 'dev' into AH-228

Anders-Hopkins committed:
Full table loads at page load

Removed the line that blocks the station table from loading before a parameter is selected. Load the full station table loads when the page is opened.

kjacobsen16 reviewed (approved):
Looks and works great! I think when we get further in, we'll probably restructure a bit. For example, the get stations call can be one function, and the agencies and stationtypes will be optionally added to it so we can have stations filtered by both of those (rather than having separate getstationsbytype and getstationsbyagency functions) and we might want to create a separate "gagestats.service.ts" for the gagestats calls, but this is a great start!
